### ISSUE TITLE
Issue with Parsing SELECT Clause Containing Equality Check (col_1 = 0)

### ISSUE THREAD

**Issue by k-ohmori** [taozhi8833998/node-sql-parser#1642]
**Describe the bug**
When parsing SQL queries that contain an equality check (col_1 = 0) with alias in the SELECT clause, node-sql-parser throws an error.

**Database Engine**
MySQL 8.0

**To Reproduce**

```
const { Parser } = require('node-sql-parser');
const parser = new Parser();
const sql = 'SELECT col_1 = 0 AS is_admin FROM sample_table;';
const ast = parser.astify(sql);

s: Expected "#", "%", "&", "*", "+", ",", "-", "--", "/", "/*", ";", "<<", ">>", "FOR", "FROM", "GO", "GROUP", "HAVING", "INTERSECT", "INTO", "LIMIT", "LOCK", "MINUS", "ORDER", "UNION", "WHERE", "WINDOW", "^", "div", "|", "||", "~", [ \t\n\r], or end of input but "A" found.
```

**Expected behavior**
I would expect the parser to correctly interpret the SQL query and generate an appropriate AST. 
The following query works fine.
```
SELECT (col_1 = 0) AS is_admin FROM sample_table;
```

**Comment by taozhi8833998** [comment]
@k-ohmori it has been fixed already and will be available in the 4.12.0

**Comment by k-ohmori** [comment]
great! thank you.
